lib/livebook/config.ex

```diff
@@ -4,21 +4,21 @@ defmodule Livebook.Config do
   @type auth_mode() :: :token | :password | :disabled
 
   @doc """
-  Checks if the distribution mode is configured to use short names.
+  Returns the longname if the distribution mode is configured to use long names.
   """
-  @spec shortnames?() :: boolean()
-  def shortnames?() do
-    case Application.get_env(:livebook, :node) do
-      nil -> true
-      {:shortnames, _name} -> true
-      {:longnames, _name} -> false
+  @spec longname() :: binary() | nil
+  def longname() do
+    [_prefix, suffix] = node() |> Atom.to_string() |> :binary.split("@")
```

Comment:
There's a similar `host` computation in `Utils` that we could unify:

https://github.com/elixir-nx/livebook/blob/00edceafc9026ba1490b5032ccd3070dd57465b9/lib/livebook/utils.ex#L39